game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/SupportCalculator.java

```diff
@@ -0,0 +1,89 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+import org.triplea.java.collections.IntegerMap;
+
+/** Calculates how much support units can give */
+@Value
+@Getter(value = AccessLevel.PACKAGE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class SupportCalculator {
```

Comment:
Is the code in this class all just moved code? Anything new or modified here?

Reply:
The only change is supportRules is now `Map<UnitSupportAttachment.BonusType, List<UnitSupportAttachments>>` instead of `List<List<UnitSupportAttachments>>`

Reply:
Thanks :+1: 

Any thoughts to use multimap here?

Guava's multimap is a data structure to represent `Map<X, List<Y>>`

Reply:
Good idea.  I didn't want to make too big of a change to this code, though.  I want to get better tests around it and then try and make the code easier to understand.